### DISCUSSION TITLE
No input element found. Please make sure that the "inputRef" template ref is properly assigned.

### DISCUSSION THREAD

**Discussion by zanzara** [dm4t2/vue-currency-input#303] [Q&A] [answered]
I get this error, if my `q-input  ref="inputRef"`  is conditionally not renderred by a `v-if` clause .
```js
 <template v-if="mode'">
<q-input
   v-model="model"
   ref="inputRef"
...
...
setup() {
const { inputRef , setOptions ,setValue } = useCurrencyInput(currencyOptions(state.locale))

return {
 inputRef,
 ....
 }
```
how could I supress/avoid this error?

**Comment by dm4t2** [comment]
A workaround would be to use `v-if` in the parent component.

**Reply by zanzara** [reply]
the issue is, that I cannot v-if the parent, because my component is pretty much used in more cases.
In the real code of my example above, there exist a v-else to the tempate.

this is now my workaround so far. Not nice, but it works.
```js
  <q-input
       v-if="!mode'"
       v-show="false"
       ref="inputRef"
   />
```

Maybe you should think about such a use-case in further releases.
But anyway: good lib. thx for sharing it

**Reply by dm4t2** [reply]
I have created an issue: https://github.com/dm4t2/vue-currency-input/issues/304

**Comment by dm4t2** [comment, marked as answer]
Fixed in [v2.3.0](https://github.com/dm4t2/vue-currency-input/releases/tag/2.3.0) 🎉

**Reply by zanzara** [reply]
Thanks. With 2.3.0 it works now also w/o my workaround.
